regnety/models/model.py

```diff
@@ -0,0 +1,90 @@
+import tensorflow as tf
+
+from regnety.regnety.config import get_model_config, ALLOWED_FLOPS
+from regnety.regnety.models.blocks import PreStem, Stem, Stage, Head
+
+from typing import List, Tuple, Union
+
+class RegNetY(tf.keras.Model):
+    """
+    RegNetY model class. Subclassed from tf.keras.Model. Instantiates a randomly
+    initialised model (to be changed after training). User must provide `flops`
+    argument to model.
+
+    Args:
+        flops: Flops of the model eg. "400MF" (Processing one image requires 
+            400 million floating point operations (multiplication, addition))
+        input_shape: A python list or tuple denoting the shape input. Please omit the 
+            batch dimension. eg. (256, 256, 3). Must be greater than or equal to 224.
+    """
+
+    def __init__(self, flops:str = "", input_shape: Union[List, Tuple] = None ):
+        super(RegNetY, self).__init__()
+
+        if flops not in ALLOWED_FLOPS:
+            raise ValueError("`flops` must be one of " + str(ALLOWED_FLOPS))
+        
+        if input_shape is None:
+            self.input_shape = (224,224,3)
+        else: 
+            self.input_shape = input_shape
```

Comment:
It's causing a conflict with a read-only @property of the base class.

Reply:
Sorry, forgot about that. I'll change it.